### PR TITLE
Utils.js, Utils2.js, Utils3.js, update to web3 v1.2.4

### PR DESCRIPTION
**Pull request by ayobuenavista**
I've also updated the packages.json with latest versions, and it's working for me. Please test.

### CODE REVIEW

File: test/v4/utils.js [hunk 1]
```diff
@@ -226,25 +228,25 @@ contract('utils', function(accounts) {
 function calcDestQty (srcQty, rate, srcDecimal, dstDecimal) {
```
Comment (ilanDoron) [inline test/v4/utils.js]: these function should be in helper file.
in one place for all to use...

File: test/v4/utils.js [hunk 2]
```diff
@@ -226,25 +228,25 @@ contract('utils', function(accounts) {
...
-    return result.floor();
+    return Math.floor(result)
 }
 
 function calcSrcQty(dstQty, srcDecimals, dstDecimals, rate) {
     //source quantity is rounded up. to avoid dest quantity being too low.
     let numerator;
     let denominator;
     if (srcDecimals >= dstDecimals) {
-        numerator = PRECISION.mul(dstQty).mul((new BigNumber(10)).pow(srcDecimals - dstDecimals));
```
Comment (ilanDoron) [inline test/v4/utils.js]: please move to helper file.

File: package.json [hunk 3]
```diff
@@ -20,30 +20,30 @@
   },
```
Comment (manhlx3006) [inline package.json]: This file + podlock file shouldn't be changed I think

File: test/v4/utils.js [hunk 4]
```diff
@@ -226,25 +228,25 @@ contract('utils', function(accounts) {
...
     let result;
     if (dstDecimal >= srcDecimal) {
-        result = ((((new BigNumber(srcQty)).mul(rate).mul((new BigNumber(10)).pow(dstDecimal - srcDecimal))).div(PRECISION)));
+        result = (((new BN(srcQty)).mul(rate).mul((new BN(10)).pow(new BN(dstDecimal - srcDecimal)))).div(PRECISION));
     } else {
-        result = ((new BigNumber(srcQty)).mul(rate).div(PRECISION.mul((new BigNumber(10)).pow(srcDecimal - dstDecimal))));
+        result = (new BN(srcQty)).mul(rate).div(PRECISION.mul((new BN(10)).pow(new BN(srcDecimal - dstDecimal))));
     }
-    return result.floor();
+    return Math.floor(result)
```
Comment (manhlx3006) [inline test/v4/utils.js]: Do we have `Math.floor(BN)`?

File: test/v4/utils.js (hunk 2 above)
Comment (Anyhowclick) [reply test/v4/utils.js]: Moved to helper file, please use that thanks =)

File: test/v4/helper.js [hunk 5]
```diff
@@ -202,14 +199,14 @@ module.exports.calcDstQty = function(srcQty, srcDecimals, dstDecimals, rate) {
...
 }
 
-module.exports.calcRateFromQty = function(srcAmount, dstAmount, srcDecimals, dstDecimals) {
-
+module.exports.calcRateFromQty = function(srcQty, dstQty, srcDecimals, dstDecimals) {
+    let decimals;
     if (dstDecimals >= srcDecimals) {
-            let decimals = new BN(10).pow(new BN(dstDecimals - srcDecimals));
-            return ((precisionUnits.mul(dstAmount)).div(decimals.mul(srcAmount)));
+        decimals = new BN(10).pow(new BN(dstDecimals - srcDecimals));
```
Comment (ilanDoron) [inline test/v4/helper.js]: why not use let.
in js we use let or const.

File: test/v4/utils.js [hunk 6]
```diff
@@ -1,12 +1,14 @@
 let MockUtils = artifacts.require("./mockContracts/MockUtils.sol")
```
Comment (ilanDoron) [inline test/v4/utils.js]: please change to const

File: test/v4/utils2.js [hunk 7]
```diff
@@ -3,12 +3,14 @@ const TokenNoDecimal = artifacts.require("./mockContracts/TokenNoDecimal.sol");
...
 
 const Helper = require("./helper.js");
-const BigNumber = require('bignumber.js');
+const BN = web3.utils.BN;
 
 const ethAddress = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE';
-const PRECISION = new BigNumber(10).pow(18);
-const MAX_QTY = new BigNumber(10).pow(28);
-const MAX_RATE = new BigNumber(10).pow(24);
+const lowerCaseEthAdd = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
```
Comment (ilanDoron) [inline test/v4/utils2.js]: don't see lowerCaseEthAdd used anywhere.
please remove from all files.

File: test/v4/utils3.js [hunk 8]
```diff
@@ -1,11 +1,14 @@
...
 const Helper = require("./helper.js");
-const BigNumber = require('bignumber.js');
-
-const PRECISION = new BigNumber(10).pow(18);
-const MAX_QTY = new BigNumber(10).pow(28);
-const MAX_RATE = new BigNumber(10).pow(24);
+const BN = web3.utils.BN;
+
+const ethAddress = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE';
+const lowerCaseEthAdd = '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
```
Comment (ilanDoron) [inline test/v4/utils3.js]: please remove

File: test/v4/helper.js (hunk 5 above)
Comment (ayobuenavista) [reply test/v4/helper.js]: `decimals` is earlier declared in line 203. This is so that there is only one initialization of `decimals` in the function.

Comment (ilanDoron) [reply test/v4/helper.js]: Oh
Sorry
missed that.
OK

**Pull request merged**